src/array.cr

```diff
@@ -457,14 +457,26 @@ class Array(T)
     self[*Indexable.range_to_index_and_count(range, size)]
   end
 
+  # like `#[Range(Int, Int)]`, but returns nil if the range's start is out of range.
```

Comment:
Let's capitalize on the beginning of a method doc please. Also can we put markdown on `nil`?
```suggestion
  # Like `#[Range(Int, Int)]`, but returns `nil` if the range's start is out of range.
```